### PR TITLE
Rework seed extension control

### LINKED ISSUE #1781: Seed is updated too frequently
**How to reproduce it (as minimally and precisely as possible)**:

1. Setup Seed and ensure it is ready.
```
$ k get seed foo
NAME   STATUS    PROVIDER (CLOUDPROFILE)   REGION      AGE   VERSION       K8S VERSION
foo    Ready     aws (aws)                 us-east-1   16m   v0.34.0-dev   v1.15.7
```

2. Watch the Seed and ensure that its `.metadata.resourceVersion` is updated on each 3-4 seconds.
```
$ watch -n 0.5 kubectl  get seeds foo -o yaml
```

The `AllExtensionsReady` condition `lastUpdateTime` is updated at each  3-4 seconds.

I guess it is not okay to have updates so frequently when the `.controllers.seed.syncPeriod` is `1m` and `.controllers.controllerInstallationCare.syncPeriod` is `30s`.

**Environment**:

- Gardener version: master
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

### PR DESCRIPTION
**Pull request by mvladev**
**What this PR does / why we need it**:

It now updates the status only when there is an actual change in it. Health checks on the controller installations (changes in their `.status`) no longer cause the condition `ExtensionsReady` to have it's `LastUpdateTime` constantly modified when there are no actual changes in `Reason` and `Message` fields.

This PR also adds additional helper `ConditionBuilder` in  `pkg/apis/core/v1{alpha1|beta1}/helper`. This allows for fluent update of the condition and checks to ensure that correct `LastUpdateTime` and `LastTransitionTime` are set only when the condition has been modified.

It also moves the `metav1.Now` function provided inside of the struct making it easier for tests and without any global state modifications.

**Which issue(s) this PR fixes**:

Fixes #1781

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: pkg/gardenlet/controller/seed/seed_extension_control.go [hunk 1]
```diff
@@ -17,22 +17,20 @@ package seed
 import (
 	"fmt"
 
-	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
-	gardencorev1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	api "github.com/gardener/gardener/pkg/apis/core/v1beta1"
```
Comment (rfranzke) [inline pkg/gardenlet/controller/seed/seed_extension_control.go]: Why was this renamed? Can we keep it `gardencorev1beta1`?

File: pkg/gardenlet/controller/seed/seed_extension_control_test.go [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package seed_test
+
+import (
+	"fmt"
+	"time"
+
+	api "github.com/gardener/gardener/pkg/apis/core/v1beta1"
```
Comment (rfranzke) [inline pkg/gardenlet/controller/seed/seed_extension_control_test.go]: Same like above. Can we keep this `gardencorev1beta1` everywhere?

File: pkg/gardenlet/controller/seed/seed_suite_test.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestSeed(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Seed Suite")
```
Comment (rfranzke) [inline pkg/gardenlet/controller/seed/seed_suite_test.go]:
```suggestion
	RunSpecs(t, "Seed Controller Suite")
```

**Review by rfranzke — COMMENTED** [review]
Nice PR! Also, thanks for adding a test to the controller 💯

File: pkg/gardenlet/controller/seed/seed_extension_control.go (hunk 1 above)
Comment (mvladev) [reply pkg/gardenlet/controller/seed/seed_extension_control.go]: Next time when we change the version to `v1` everything needs to be replaced.

Comment (rfranzke) [reply pkg/gardenlet/controller/seed/seed_extension_control.go]: Yes, but I still find it WAY more clear and better understandable reading the code when the API version is used as import alias (`corev1.Secret, appsv1.StatefulSet`, ...). Sometimes there are other structs with the same name like a resource, and then it's not always clear where this comes from.

**Comment by mvladev** [comment]
Rebased with recommendations. Have another look.

**Review by rfranzke — APPROVED** [review]
/lgtm, thank you @mvladev !
